lib/context.js

```diff
@@ -4,7 +4,7 @@ const { kFourOhFourContext, kReplySerializerDefault } = require('./symbols.js')
 
 // Objects that holds the context of every request
 // Every route holds an instance of this object.
-function Context (schema, handler, Reply, Request, contentTypeParser, config, errorHandler, bodyLimit, logLevel, attachValidation, replySerializer) {
+function Context (schema, handler, Reply, Request, contentTypeParser, config, errorHandler, bodyLimit, logLevel, attachValidation, replySerializer, onResponse) {
```

Comment:
I think the new constructor parameter is not required. The new hook will be added in the following line, since it was added to the list of the route level hooks:
https://github.com/fastify/fastify/blob/936b5090c39467cbebf4b212f62917e81e61e086/lib/route.js#L260-L264

Reply:
If not passed by Context, they not work, because not have referenced it

Reply:
+1 of @SerayaEryn 's review